src/Monolog/Formatter/JsonFormatter.php

```diff
@@ -145,7 +145,7 @@ protected function normalize($data, $depth = 0)
             return 'Over 9 levels deep, aborting normalization';
         }
 
-        if (is_array($data) || $data instanceof \Traversable) {
```

Comment:
This code throw fatal error only for `Generator`s . 

Can you just `is_array($data) || ($data instanceof \Traversable && !($data instanceof \Generator))`?

I guess idea to use only arrays is very bad.

Reply:
thanks, changed

Reply:
I'd tend to say rather remove this behavior entirely from Traversables, much like #891 did, otherwise there is a behavior difference between NormalizerFormatter and JsonFormatter.

Reply:
Changed again.